deepmedic/logging/accuracyMonitor.py

```diff
@@ -146,12 +148,91 @@ def reportAccuracyForLastSubepoch(self) :
-            self.log.print3(logStrClass+"\t mean sensitivity:\t"+ strFl4fNA(meanAccOnPosOfSubep, self.NA_PATTERN)+"\t=> TruePos/RealPos = "+str(numOfTpInSubep)+"/"+str(numOfRpInSubep))
-            self.log.print3(logStrClass+"\t mean precision:\t"+ strFl4fNA(meanPrecOfSubep, self.NA_PATTERN)+"\t=> TruePos/(TruePos+FalsePos) = "+str(numOfTpInSubep)+"/"+str(numOfTpInSubep+numOfFpInSubep))
-            self.log.print3(logStrClass+"\t mean specificity:\t"+ strFl4fNA(meanAccOnNegOfSubep, self.NA_PATTERN)+"\t=> TrueNeg/RealNeg = "+str(numOfTnInSubep)+"/"+str(numOfRnInSubep))
-            self.log.print3(logStrClass+"\t mean Dice:       \t"+ strFl4fNA(meanDiceOfSubep, self.NA_PATTERN))
-            
+            self.log.print3(logStrClass+"\t mean accuracy:   \t" +
+                            strFl4fNA(meanAccClassOfSubep, self.NA_PATTERN) +
+                            "\t=> (TruePos+TrueNeg)/All-Predicted-Voxels = " +
+                            str(numOfTpInSubep+numOfTnInSubep) + "/" +
+                            str(numOfRpInSubep+numOfRnInSubep))
+            self.log.print3(logStrClass+"\t mean sensitivity:\t" + strFl4fNA(meanAccOnPosOfSubep, self.NA_PATTERN)+"\t=> TruePos/RealPos = "+str(numOfTpInSubep)+"/"+str(numOfRpInSubep))
+            self.log.print3(logStrClass+"\t mean precision:\t" + strFl4fNA(meanPrecOfSubep, self.NA_PATTERN)+"\t=> TruePos/(TruePos+FalsePos) = "+str(numOfTpInSubep)+"/"+str(numOfTpInSubep+numOfFpInSubep))
+            self.log.print3(logStrClass+"\t mean specificity:\t" + strFl4fNA(meanAccOnNegOfSubep, self.NA_PATTERN)+"\t=> TrueNeg/RealNeg = "+str(numOfTnInSubep)+"/"+str(numOfRnInSubep))
+            self.log.print3(logStrClass+"\t mean Dice:       \t" + strFl4fNA(meanDiceOfSubep, self.NA_PATTERN))
+
+    def log_to_tensorboard(self, metrics_dict, class_string, step_num):
+        if self.tensorboard_logger is not None:
+            for metric, value in metrics_dict.items():
+                if value == self.NA_PATTERN:
+                    value = np.nan
+                self.tensorboard_logger.add_summary(value, metric + '_' + class_string, step_num)
+
+    def log_acc_subep_to_tensorboard(self):
+
+        trainOrValString = "TRAINING" if self.training0orValidation1 == 0 else "VALIDATION"
+        currSubep = self.numberOfSubepochsForWhichUpdated - 1
+
+        self.log.print3('=============== LOGGING TO TENSORBOARD ===============')
+        self.log.print3('Logging ' + trainOrValString + ' metrics')
+        self.log.print3('Epoch: ' + str(self.epoch) +
+                        ' | Subepoch ' + str(currSubep) + '/' + str(self.numberOfSubepochsPerEpoch - 1))
+        step_num = currSubep + (self.epoch * (self.numberOfSubepochsPerEpoch))
+        self.log.print3('Step number: ' + str(step_num))
+
+        # check if user included tensorboard logging in the config
+        if self.tensorboard_logger is None:
```

Comment:
I 'd put this right under the "logging to tensorboard", to avoid printing epoch/subepoch etc if it's deactivated. Less logging. (eg "step number" does not make sense).